File: Samples/Magick.NET.Samples/Samples/Magick.NET/DetailedDebugInformation.vb
' Copyright 2013-2018 Dirk Lemstra <https://github.com/dlemstra/Magick.NET/>
'
' Licensed under the ImageMagick License (the "License"); you may Not use this file except in
' compliance with the License. You may obtain a copy of the License at
'
'   https://www.imagemagick.org/script/license.php
'
' Unless required by applicable law Or agreed to in writing, software distributed under the
' License Is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES Or CONDITIONS OF ANY KIND,
' either express Or implied. See the License for the specific language governing permissions
' And limitations under the License.

Imports System
Imports ImageMagick

Namespace RootNamespace.Samples.MagickNET

    Public NotInheritable Class DetailedDebugInformationSamples

        Private Shared Sub WriteLogMessage(message As String)
            Console.WriteLine(message)
        End Sub

        Public Shared Sub MagickNET_Log(sender As Object, arguments As LogEventArgs)
            ' Write log message
            WriteLogMessage(arguments.Message)
        End Sub

        Public Shared Sub ReadImage()
            ' Log all events
            ImageMagick.MagickNET.SetLogEvents(LogEvents.All)
            ' Set the log handler (all threads use the same handler)
            AddHandler ImageMagick.MagickNET.Log, AddressOf MagickNET_Log

            Using image As New MagickImage()
                ' Reading the image will send all log events to the log handler
                image.Read(SampleFiles.SnakewarePng)
            End Using
        End Sub

    End Class

End Namespace

File: samples/Magick.NET.Samples/DetailedDebugInformation.cs
// Copyright Dirk Lemstra https://github.com/dlemstra/Magick.NET.
// Licensed under the Apache License, Version 2.0.

using System;
using ImageMagick;

namespace Magick.NET.Samples;

public static class DetailedDebugInformationSamples
{
    public static void MagickNET_Log(object sender, LogEventArgs arguments)
    {
        // Write log message
        if (arguments.Message is not null)
        {
            Console.WriteLine(arguments.Message);
        }
    }

    public static void ReadImage()
    {
        // Log all events
        MagickNET.SetLogEvents(LogEvents.All);

        // Set the log handler (all threads use the same handler)
        MagickNET.Log += MagickNET_Log;

        using var image = new MagickImage();

        // Trace logging checks if this is set to true.
        image.Settings.Debug = true;

        // Reading the image will send all log events to the log handler
        image.Read(SampleFiles.SnakewarePng);
    }
}
